apps/browser/qml/pages/HistoryPage.qml

```diff
@@ -35,6 +37,14 @@ Page {
             root.loadPage(url)
         }
 
+        onOpenInPrivateTab: {
```

Comment:
Just a suggestion, but rather than duplicate the code, it might be worth checking whether it would be neater to extend `onLoad` with an optional `private` bool parameter (I've not tried it, so it might also end up complicating things: your judgement).

Reply:
```onLoad``` if only you add the functionality of opening a new tab to it ...

Reply:
I was thinking that the logic about whether to open a new tab might also be simplified if it were pushed up into Overlay.loadPage().

Reply:
Done